File: 06- Stack/C#/1- Removing Stars From a String.cs
public class Solution {
    public string RemoveStars(string s) {
        Stack<char> st = new Stack<char>();

        foreach(char c in s){
            if (c != '*')
                st.Push(c);
            else if (st.Count != 0 && c == '*')
                st.Pop() ;
        }
        char[] ans = new char[st.Count] ; 
        for (int i = st.Count - 1 ; i >= 0 ; i--)
            ans[i] = st.Pop() ; 

        return new string(ans); 
    }
}

File: 06- Stack/1- Removing Stars From a String.cpp
class Solution {
public:
    string removeStars(string s) {
        stack<char> st ;
        for (auto c : s) {
            if (c != '*')
                st.push(c) ;
            else if (!st.empty() && c == '*')
                st.pop();
        }
        string ans = "" ;
        while (!st.empty()) {
            ans += st.top();
            st.pop();
        }
        reverse(ans.begin() , ans.end());
        return ans ;
    }
};